initializr-generator/src/main/java/io/spring/initializr/generator/project/ProjectDescriptionModifications.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.initializr.generator.project;
+
+import io.spring.initializr.generator.buildsystem.BuildSystem;
+import io.spring.initializr.generator.version.Version;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+
+public class ProjectDescriptionModifications {
+
+	private MutableProjectDescription original;
+	private MutableProjectDescription copy;
+
+	public ProjectDescriptionModifications(final MutableProjectDescription original) {
+		
+		final MutableProjectDescription copy = new MutableProjectDescription();
+		copy.setPlatformVersion(original.getPlatformVersion());
+		copy.setBuildSystem(original.getBuildSystem());
+		copy.setPackaging(original.getPackaging());
+		copy.setLanguage(original.getLanguage());
+		original.getRequestedDependencies().forEach(copy::addDependency);
+		copy.setGroupId(original.getGroupId());
+		copy.setArtifactId(original.getArtifactId());
+		copy.setVersion(original.getVersion());
+		copy.setName(original.getName());
+		copy.setDescription(original.getDescription());
+		copy.setApplicationName(original.getApplicationName());
+		copy.setPackageName(original.getPackageName());
+		copy.setBaseDirectory(original.getBaseDirectory());
+
+		this.original = original;
+		this.copy = copy;
```

Comment:
For some reason that makes me a bit nervous but I can't really say why. Passing the reference to the diff before it's going to be mutated and rely on the reference looks wrong (although it simplifies other things).

This is going to fall down once we look at the custom `ProjectDescription` use case.

Reply:
K. Will keep an eye on it when I do.